zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobUpdateProcessor.java

```diff
@@ -59,15 +57,20 @@ public void processRecord(final TypedRecord<JobRecord> command) {
     if (retries > 0) {
       job.setRetries(retries);
       stateWriter.appendFollowUpEvent(key, JobIntent.RETRIES_UPDATED, job);
-      responseWriter.writeEventOnCommand(key, JobIntent.RETRIES_UPDATED, job, command);
+      isJobUpdated = true;
     }
     // Handle timeout
     final long oldDeadline = job.getDeadline();
     if (timeout > -0 && jobState.jobDeadlineExists(key, oldDeadline)) {
       final long newDeadline = ActorClock.currentTimeMillis() + job.getTimeout();
       job.setDeadline(newDeadline);
       stateWriter.appendFollowUpEvent(key, JobIntent.TIMEOUT_UPDATED, job);
-      responseWriter.writeEventOnCommand(key, JobIntent.TIMEOUT_UPDATED, job, command);
+      isJobUpdated = true;
+    }
+
+    if (isJobUpdated) {
+      stateWriter.appendFollowUpEvent(key, JobIntent.UPDATED, job);
+      responseWriter.writeEventOnCommand(key, JobIntent.UPDATED, job, command);
```

Comment:
👍 I like that we're now just writing one response.

❓ We are also writing the follow-up event UPDATED but there is no applier to handle it. Will it simply be ignored? Is there a specific purpose for writing this event? Just wondering if we're unnecessarily writing events to the log that need to be processed/replayed but are effectively not doing anything. Seems like a waste of resources.

Reply:
Sorry I missed a piece 🙇 

We always append the response as a event in the state for data consistency and debugging, in this case the response of the UPDATED event will be handled by a NoopApplier.

Reply:
OK, I see, sounds plausible.